pkg/cloudprovider/providers/azure/azure_util.go

```diff
@@ -116,6 +125,204 @@ func (az *Cloud) getSecurityRuleID(securityRuleName string) string {
 		securityRuleName)
 }
 
+// returns the full identifier of a publicIPAddress.
+func (az *Cloud) getpublicIPAddressID(pipName string) string {
+	return fmt.Sprintf(
+		publicIPAddressIDTemplate,
+		az.SubscriptionID,
+		az.ResourceGroup,
+		pipName)
+}
+
+// select load balancer for the service in the cluster
+// the selection algorithm selectes the the load balancer with currently has
```

Comment:
I am probably misunderstanding Azure load balancers here but when I was nosing around this before I found two limitations:

* A VM can be associated with at most one external and one internal LB
* A LB cannot back onto VMs from more than one availability set

Given these limitations, I'm not sure how a service can get to choose which LB it is exposed on.  Surely it _has_ to surface on the LB associated with its availability set.  I'm clearly missing something, because you've obviously tested this and proved that it works - what am I not getting?

Reply:
the kubernetes nodes proxy and route the request to the correct node. a service lb can therefore run on any other availability set (than the one its running on)

Reply:
Oh wow, that's very neat! Thank you, I didn't know that!